### PR TITLE
fix:Temporarily disable Discount code POST test to fix build

### LINKED ISSUE #4853: Temporarily disable Discount code post test
**Is your feature request related to a problem? Please describe.**

currently the build is failing due to an unknown error arising due to the discount code post request.

**Describe the solution you'd like**
For now, let's skip the test untill we figure out the problem so that it doesn't hamper the progress of the project.

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #4853

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
currently the build is failing due to an unknown error arising due to the discount code post request.

#### Changes proposed in this pull request:
For now, skipping the test untill we figure out the problem so that it doesn't hamper the progress of the project.

### CODE REVIEW

**Comment by srv-twry** [comment]
@niranjan94 @SaptakS @iamareebjamal Please review and merge. 
I understand that this is a workaround and isn't the recommended way but I think we should let this issue halt the project's development.

File: tests/hook_main.py [hunk 1]
```diff
@@ -2389,10 +2389,12 @@ def event_discount_code_post(transaction):
     :param transaction:
     :return:
     """
-    with stash['app'].app_context():
-        event = EventFactoryBasic()
-        db.session.add(event)
-        db.session.commit()
+    transaction['skip'] = True
```
[the remark this answers is not in the document]
Comment (srv-twry) [reply tests/hook_main.py]: done.

Comment (schedutron) [inline tests/hook_main.py]: What does `transaction['skip'] = True` do?

Comment (srv-twry) [reply tests/hook_main.py]: skips the test

**Pull request merged**